pkg/controller/controller.go

```diff
@@ -53,6 +54,9 @@ const (
 	//
 	pollingStartInterval      = 1 * time.Second
 	pollingMaxBackoffDuration = 1 * time.Hour
+
+	// amount of time to allow the reconciler to retry operations
+	retryDuration = 7 * 24 * time.Hour
```

Comment:
Can we make this configurable via some config file or cmd line flag?

Reply:
I have added a `--reconciliation-retry-duration` command-line flag that can be used to set the retry duration to something other than the default of 7 days.